Darwin linker on x86 complains like

../swipl-ld.sh -shared -o snowball.dylib snowball.o
libstemmer_c/libstemmer.a
ld: in libstemmer_c/libstemmer.a, archive has no table of contents
collect2: ld returned 1 exit status
i686-apple-darwin9-gcc returned code 256

So just make sure we have an index in the archive (-s).

--- packages/nlp/libstemmer_c/Makefile.pl
+++ packages/nlp/libstemmer_c/Makefile.pl
@@ -4,7 +4,7 @@
 all: libstemmer.a
 
 libstemmer.a: $(snowball_sources:.c=.o)
-	$(AR) -cru $@ $^
+	$(AR) -scru $@ $^
 stemwords: examples/stemwords.o libstemmer.a
 	$(CC) $(CFLAGS) -o $@ $^
 
